### PR TITLE
Merge Product Compatibility Matrix into Docs

### PR DESCRIPTION
**Pull request by lucasvaltl**
## Description
- This PR merges the product compatibility matrix we have in notion into our documentation. The aim is for this new docs page to be the source of truth for what Gitpod can and cannot work with. In order to capture the responsible team, I added a small commented out table at the bottom that captures this info. This should be more readable.
- In addition, the required components page has been changed in its intent. It moved to the background section and is now aimed at giving an understanding of what dependencies Gitpod has and why. It does not include the version or type of dependencies that are supported - instead it refers to the product compatibility page for this.

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/10126

## How to test
1. Review product compatibility page: Open netlify preview, navigate to docs>self-hosted>reference>product compatibility
2. Review changes to required components page: Open netlify preview, navigate to docs>self-hosted>background>dependencies

## Release Notes

```release-note
NONE
```

<a href="https://gitpod-staging.com/#https://github.com/gitpod-io/website/pull/2433"></a>

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/2433"></a>

### CODE REVIEW

File: gitpod/docs/self-hosted/latest/required-components.md [hunk 1]
```diff
@@ -1,69 +1,38 @@
...
+subsection: background
+title: Components & Dependencies
 ---
 
 <script context="module">
   export const prerender = true;
 </script>
 
-# Required Components
+# Gitpod Self-Hosted ependencies
```
Comment (MrSimonEmms) [inline gitpod/docs/self-hosted/latest/required-components.md:11]: [D's are good](https://www.youtube.com/watch?v=YFJdUJg4wOk&ab_channel=slycro)

Comment (lucasvaltl) [inline gitpod/docs/self-hosted/latest/required-components.md:11]:
```suggestion
# Gitpod Self-Hosted Dependencies
```

Comment (lucasvaltl) [reply gitpod/docs/self-hosted/latest/required-components.md]: :D

**Comment by csweichel** [comment]
Love it - this goes well and truly in the right direction.
A few notes:
- on the product compatibility page, I'd use `rowspan` to group the categories. With a fixed stylesheet this makes it easier to read: image
- reference architectures should IMHO become a point under `Reference -> Architectures`
- I'd move the "Background -> Dependencies" page under `Reference`

File: src/lib/contents/docs/menu.ts [hunk 2]
```diff
@@ -117,8 +117,11 @@ export const MENU: MenuEntry[] = [
       M("Support Bundles", "self-hosted/latest/support-bundle", []),
       M("Support Patches", "self-hosted/latest/support-patches", []),
     ]),
+    M("Background", "self-hosted/latest/background", [
```
Comment (loujaybee) [inline src/lib/contents/docs/menu.ts:120]: Does this want to be nested under self-hosted or somewhere shared in the docs? 

Also thinking about overlap to: https://github.com/gitpod-io/website/issues/2185

**Comment by lucasvaltl** [comment]
Discussing the process of howe we want to align on changes to this matrix in the future in this internal [RFC](https://www.notion.so/7499887c0d9c40a19c61c49868ef60d3). Once we aligned on that, we will use what we aligned on to progress this PR :)

**Comment by nisarhassan12** [comment]
@lucasvaltl I'm closing this Please re-open it if it is still relevant. Thanks

**Pull request closed without merging**